[ios][router] Fix NativeTabs `icon` and `selectedIcon` type mismatch crash (#48302)
expo/expo: docs/pages/router/advanced/native-tabs.mdx

            default: require('../assets/setting_icon.png'),
            selected: require('../assets/selected_setting_icon.png'),
          }}
          // So `tintColor` also applies to the selected image icon on iOS

>           renderingMode="template"

Reviewer: ```suggestion
```

Author: Done in 4c1cc8b — dropped from both `tintColor` examples. The shared-rendering-mode caveat is still covered in the Known limitations section.

            default: require('../assets/setting_icon.png'),
            selected: require('../assets/selected_setting_icon.png'),
          }}
          // So `tintColor` also applies to the selected image icon on iOS

>           renderingMode="template"

Reviewer: ```suggestion
```

Author: rn

Author: give me 20 min, will push soon

Author: Done in 4c1cc8b.

> **info** The `renderingMode` prop only affects iOS. On Android, all image icons are rendered with their original colors.

> > **info** On iOS, a tab's default and selected icons must use the same rendering mode. When they resolve to different modes, for example when only `selectedIconColor` is set, both icons use the default icon's rendering mode and a warning is logged.

Reviewer: Can you please move this information to Known Limitations section?

## Known limitations

<Collapsible summary="The default and selected icons share one rendering mode on iOS">

> On iOS, a tab's default and selected image icons must use the same [rendering mode](#icon-rendering-mode). When they resolve to different modes, both icons use the default icon's mode and a warning is logged.

Reviewer 2: ✏️ **style** `prose.md#voice-and-tone`

"a warning is logged" is passive and leaves out who logs it and when. Name the actor: "Expo Router logs a warning in development".

```suggestion
On iOS, a tab's default and selected image icons must use the same [rendering mode](#icon-rendering-mode). When they resolve to different modes, both icons use the default icon's mode and Expo Router logs a warning in development.
```

<!-- wm-review -->

> The modes disagree when an icon color applies to only one of the states. This happens when you set `tintColor`, `iconColor={{ selected }}`, or the `Icon` `selectedColor` prop without also setting a color for the default state, because a color implies `'template'` rendering while an uncolored icon defaults to `'original'`. Setting `renderingMode` for only one state has the same effect.

Reviewer 2: ✏️ **style** `mechanics.md#vale-sentencelength`

The sentence starting "This happens when you set" runs past 25 words. Split it after "the default state" so the explanation stands as its own sentence.

```suggestion
The modes disagree when an icon color applies to only one of the states. This happens when you set `tintColor`, `iconColor={{ selected }}`, or the `Icon` `selectedColor` prop without also setting a color for the default state. A color implies `'template'` rendering, while an uncolored icon defaults to `'original'`. Setting `renderingMode` for only one state has the same effect.
```

<!-- wm-review -->
